### PR TITLE
R46 add service worker

### PR DESCRIPTION
**Pull request by RobertoMaurizzi**
Closes https://github.com/catalpainternational/openly_dird/issues/2040

# Description
Adds a service worker to the `/embed/` path of Metabase, used when serving embedded dashboards.
The SW will cache anything under `/app/dist/` if it's in the format `<filename>.<extension>?<20_digit_hex_string>` that is the format Metabase uses to version and cache-bust its asset.

More details about Metabase and its application architecture and development can be found in https://github.com/catalpainternational/openly_dird/issues/2040

### CODE REVIEW

File: frontend/src/metabase/sw.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import {
+  StaleWhileRevalidate,
+  NetworkFirst,
+  NetworkOnly,
+} from "workbox-strategies";
+
+// network first for HTML pages
+registerRoute(
+  ({ url }) => url.pathname === "/" || url.pathname === "/app/",
+  new NetworkFirst(),
```
Comment (raphaelmerx) [inline frontend/src/metabase/sw.js]: I'm realising with some extra work, we'd be able to have that embedded dashboard load offline, which is a big deal! Would be very useful in the districts.

File: package.json [hunk 2]
```diff
@@ -105,6 +105,7 @@
     "tippy.js": "^6.3.5",
     "ttag": "1.7.15",
     "underscore": "~1.13.3",
+    "workbox-webpack-plugin": "^7.0.0",
```
Comment (raphaelmerx) [inline package.json]: Probably need to add `workbox-strategies` and `workbox-routing` here, since we use them directly in sw.js

Comment (RobertoMaurizzi) [reply package.json]: They're indirect dependencies and got installed and listed in the yarn.lock file automatically (and I'd hope to get an error if something I import is missing? Asking for real, Webpack and Workbox looks to me a bit like Asterix's "La maison qui rend fou"...)

Comment (raphaelmerx) [reply package.json]: They're direct dependencies of our project since we use them in `sw.js`, so we should add them to package.json

**Comment by RobertoMaurizzi** [comment]
I re-tested the latest version locally too and it both builds and works for the /embed page (tested within DIRD too).
I'd say it looks like it's able to work offline too, but I didn't test that.

**Review by raphaelmerx — APPROVED** [review]

**Pull request merged**